### ISSUE TITLE
SA1118 false positive with multi-line collection expressions as arguments

### ISSUE THREAD

**Issue by ecampidoglio** [DotNetAnalyzers/StyleCopAnalyzers#3732]
Passing a multi-line [collection expression ](https://learn.microsoft.com/en-us/dotnet/csharp/whats-new/csharp-12#collection-expressions)as argument in C# 12 triggers [`SA1118`](https://github.com/DotNetAnalyzers/StyleCopAnalyzers/blob/master/documentation/SA1118.md):

```csharp
// ⚠️ SA1118 The parameter spans multiple lines
foo.Bar(
    [
        "foo",
        "bar",
        "baz"
    ]);
```

Specifying the elements on the same line does not trigger the warning:

```csharp
foo.Bar(["foo", "bar", "baz"]);
```

This is on version `1.2.0-beta.507`.

**Comment by bjornhellander** [comment]
There is currently no way of adding tests for c# 12 constructs in this repository, so this will have to wait for a while. It should be trivial to fix once that is in place, though.

Side note: Based on the analyzer code, I understand that this **should** happen, but it doesn't for me. I actually can't get SA1118 to trigger on anything. So weird. Would be interesting to hear if anyone else is having the same problem as me. Otherwise I assume that something is wrong with my computer.
